test/mocks/v1/non_template_tests_test.go

```diff
@@ -0,0 +1,194 @@
+		Expect(err).NotTo(HaveOccurred())
+		// ClusterResource Constructor
+		clusterResourceClientFactory := &factory.KubeResourceClientFactory{
+			Crd:         ClusterResourceCrd,
+			Cfg:         cfg,
+			SharedCache: kuberc.NewKubeCache(context.TODO()),
+		}
+
+		err = helpers.AddAndRegisterCrd(ctx, ClusterResourceCrd, clientset)
+		Expect(err).NotTo(HaveOccurred())
+
+		clusterResourceClient, err = NewClusterResourceClient(ctx, clusterResourceClientFactory)
+		Expect(err).NotTo(HaveOccurred())
+		// MockCustomType Constructor
+		mockCustomTypeClientFactory := &factory.MemoryResourceClientFactory{
+			Cache: memory.NewInMemoryResourceCache(),
+		}
+
+		mockCustomTypeClient, err = NewMockCustomTypeClient(ctx, mockCustomTypeClientFactory)
+		Expect(err).NotTo(HaveOccurred())
+		// Pod Constructor
+		podClientFactory := &factory.MemoryResourceClientFactory{
+			Cache: memory.NewInMemoryResourceCache(),
+		}
+
+		podClient, err = github_com_solo_io_solo_kit_pkg_api_v1_resources_common_kubernetes.NewPodClient(ctx, podClientFactory)
+		Expect(err).NotTo(HaveOccurred())
+		emitter = NewTestingEmitter(simpleMockResourceClient, mockResourceClient, fakeResourceClient, anotherMockResourceClient, clusterResourceClient, mockCustomTypeClient, podClient)
+	})
+	AfterEach(func() {
+		kubeutils.DeleteNamespacesInParallelBlocking(ctx, kube, namespace1, slowWatchNamespace)
+	})
+
+	It("Should not overwrite initial listed resources for non-returned namespace watchers", func() {
+		/*
+			This test relies on some hardcoded logic in `resource_client_template.go`.  In particular:
+			`FakeResourceClient`, when used in conjunction with the `slowWatchNamespace` has an artificial
+			5s delay when getting back the results from .Watch.  The idea behind the test is that
+			we have a normal, unimpeded watch running in parallel with a slow watch to simulate a race condition.
+		*/
+		ctx := context.Background()
+		err := emitter.Register()
+		Expect(err).NotTo(HaveOccurred())
+
+		// start by creating the test-specific namespace
+
+		err = kubeutils.CreateNamespacesInParallel(ctx, kube, slowWatchNamespace)
+		Expect(err).NotTo(HaveOccurred())
+
+		// then, write `FakeResource`s to "namespace1" and "slowWatchNamespace"
+		fakeResource1a, err := fakeResourceClient.Write(NewFakeResource(namespace1, name1), clients.WriteOpts{Ctx: ctx})
+		Expect(err).NotTo(HaveOccurred())
+		fakeResource1b, err := fakeResourceClient.Write(NewFakeResource(slowWatchNamespace, name1), clients.WriteOpts{Ctx: ctx})
+		Expect(err).NotTo(HaveOccurred())
+
+		// create a snapshot observer on "namespace1" and "slowWatchNamespace"
+		snapshots, _, err := emitter.Snapshots([]string{namespace1, slowWatchNamespace}, clients.WatchOpts{
+			Ctx:         ctx,
+			RefreshRate: time.Second,
+		})
+		Expect(err).NotTo(HaveOccurred())
+
+		timeout := time.After(time.Second * 10)
+		ticker := time.Tick(time.Second / 10)
+	outer:
+		for {
+			select {
+			case <-ctx.Done():
+				break outer
+			case <-timeout:
```

Comment:
I like this! Another alternative would be to create resources in the before each and clean them up in an after each. Currently we have some duplicate code where resources (namesapces) are cleaned up in the after each. Can we pick one pattern and use it?